src/bin/main.rs

```diff
@@ -136,12 +172,51 @@ fn check(args: &ArgMatches) -> HowserResult<Option<ValidationProblem>> {
     }
 }
 
+fn process_pharmacy_file(args: &ArgMatches) -> HowserResult<Vec<ValidationProblem>> {
+    let mut report: Vec<ValidationProblem> = Vec::new();
+
+    if let Some(filename) = args.value_of("pharmacy") {
+        if let Ok(pharmacy_contents) = get_file_contents(filename) {
+            if let Some(specs) = pharmacy_contents.parse::<Value>()?["Specs"].as_table() {
+                for rx_file in specs.keys() {
+                    if let Some(doc_file) = specs.get(rx_file).map_or(None, |value| value.as_str())
+                    {
+                        let validate_args = vec!["howser", "validate", rx_file, doc_file];
+                        let app = make_app();
```

Comment:
Essentially making an external call to the application's CLI rather than directly using its internal functions after the code flow has progressed past the initial arg parsing is fragile and inadvisable. It seems to suggest that the internal surfaces of the codebase are inaccessible to the program itself by treating its own capabilities as a black box.

Please change this to a straight function call to the appropriate validation subroutine without looping back through the CLI.